flexmeasures/data/models/planning/linear_optimization.py

```diff
@@ -307,8 +312,9 @@ def cost_function(m):
         costs = 0
         for c in m.c:
             for j in m.j:
-                costs += m.commitment_downwards_deviation[c, j] * m.down_price[c, j]
-                costs += m.commitment_upwards_deviation[c, j] * m.up_price[c, j]
+                for i in range(0,len(m.down_price[c,j])):
```

Comment:
Now that we have created a new dimension for the commitments and the prices, we can index them by the variable of the model d:

```python
for d in m.d:
     costs += m.commitment_downwards_deviation[c, j,d] * m.down_price[c, j,d]
     costs += m.commitment_upwards_deviation[c, j, d] * m.up_price[c, j,d]
```